### PR TITLE
Web3Modal + Wagmi v2 Example in Next.js App Router

### PR DESCRIPTION
**Pull request by glitch-txs**
- Added Next and React examples for Wagmi v2

### CODE REVIEW

**Comment by glitch-txs** [comment]
is the test supposed to fail? cc @arein @ganchoradkov

File: dapps/web3modal/react/package.json [hunk 1]
```diff
@@ -10,11 +10,12 @@
     "preview": "vite preview"
   },
   "dependencies": {
-    "@web3modal/wagmi": "^3.4.0",
+    "@tanstack/react-query": "^5.17.19",
+    "@web3modal/wagmi": "4.0.0-alpha.2",
```
Comment (glitch-txs) [inline dapps/web3modal/react/package.json:14]:
```suggestion
    "@web3modal/wagmi": "4.0.0-alpha.3",
```

File: dapps/web3modal/next/biome.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+		}
+	},
+	"javascript": {
+    "formatter": {
+      "semicolons": "asNeeded",
+			"quoteStyle":"single",
+			"lineWidth": 120
+    }
+	}
+}
```
Comment (tomiir) [inline dapps/web3modal/next/biome.json:19]: what's with this lint?

**Review by tomiir — APPROVED** [review]

File: dapps/web3modal/next/biome.json (hunk 2 above)
Comment (glitch-txs) [reply dapps/web3modal/next/biome.json:19]: very opinionated of mine, should I remove it? :v

File: dapps/web3modal/next/package.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    "lint": "next lint",
+    "format": "pnpm dlx @biomejs/biome format ./src --write"
+  },
+  "dependencies": {
+    "@tanstack/react-query": "^5.17.19",
+    "@web3modal/wagmi": "4.0.0-alpha.3",
+    "next": "14.0.4",
+    "react": "^18",
+    "react-dom": "^18",
+    "viem": "^2.5.0",
```
Comment (ignaciosantise) [inline dapps/web3modal/next/package.json:18]: lock the versions for the next PR 👍 

(remove the ^)

File: dapps/web3modal/react/package.json [hunk 4]
```diff
@@ -10,11 +10,12 @@
...
   "dependencies": {
-    "@web3modal/wagmi": "^3.4.0",
+    "@tanstack/react-query": "^5.17.19",
+    "@web3modal/wagmi": "4.0.0-alpha.3",
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
-    "viem": "^1.19.11",
-    "wagmi": "^1.4.8"
+    "viem": "^2.5.0",
+    "wagmi": "^2.5.2"
```
Comment (ignaciosantise) [inline dapps/web3modal/react/package.json:18]: remove those ^

**Pull request merged**